6.5.group_by.py

```diff
@@ -0,0 +1,8 @@
+def group_by(func, iterable):
+    mydict = {}
+    for part in iterable:
+        mydict.setdefault(func(part), []).append(part)
```

Comment:
You should aim to write everything in a single line, even if you can. 
This link https://towardsdatascience.com/you-should-stop-craving-for-python-one-liners-5dbf8e0de23d explains it well, even though I disagree with how important memory and runtime are in most python programs.

Reply:
I don't know how to write this in one line